File: logipack-hub/hub-web/src/lib/server/hubApi/dto/clients.ts
export type ClientDto = {
	id: string;
	name: string;

	email?: string | null;
	phone?: string | null;

	createdAt?: string | null;
	updatedAt?: string | null;
	deletedAt?: string | null;
};

export type ClientListItemDto = {
	id: string;
	name: string;
	email?: string | null;
	phone?: string | null;

	updated_at?: string | null;
};

export type ListClientsResponseDto = {
	clients: ClientListItemDto[];
};

export type GetClientResponseDto = {
	client: ClientDto;
};

export type CreateClientRequestDto = {
	name: string;
	email?: string | null;
	phone?: string | null;
};

export type CreateClientResponseDto = {
	client: ClientDto;
};

export type UpdateClientRequestDto = {
	name?: string | null;
	email?: string | null;
	phone?: string | null;
};

export type UpdateClientResponseDto = {
	client: ClientDto;
};

File: logipack-hub/hub-api/src/dto/clients.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Serialize, Deserialize)]
pub struct ClientDto {
    pub id: String,
    pub name: String,
    pub email: Option<String>,
    pub phone: Option<String>,

    pub updated_at: String,
}

#[derive(Debug, Serialize, Deserialize)]
pub struct CreateClientRequest {
    pub name: String,
    pub email: Option<String>,
    pub phone: Option<String>,
}

#[derive(Debug, Serialize, Deserialize)]
pub struct CreateClientResponse {
    pub client: ClientDto,
}

#[derive(Debug, Serialize, Deserialize)]
pub struct ListClientsResponse {
    pub clients: Vec<ClientDto>,
}

#[derive(Debug, Serialize, Deserialize)]
pub struct GetClientResponse {
    pub client: ClientDto,
}

#[derive(Debug, Serialize, Deserialize)]
pub struct UpdateClientRequest {
    pub name: Option<String>,
    pub email: Option<String>,
    pub phone: Option<String>,
}

#[derive(Debug, Serialize, Deserialize)]
pub struct UpdateClientResponse {
    pub client: ClientDto,
}
